### PR TITLE
Adds opt-in multiline parameters rule

### PR DESCRIPTION
**Pull request by ornithocoder**
This commit introduces the opt-in rule that checks if methods and functions parameters are all on the same line or one per line. For instance, the dummy function below violates the rule.

```swift
func foo(param1: Int,
         param2: Int, param3: Int) -> (Int) -> Int {
    return { x in x + param1 + param2 + param3 }
}
```

It should be implemented either as:

```swift
func foo(param1: Int, param2: Int, param3: Int) -> (Int) -> Int {
    return { x in x + param1 + param2 + param3 }
}
```

or

```swift
func foo(param1: Int,
         param2: Int,
         param3: Int) -> (Int) -> Int {
    return { x in x + param1 + param2 + param3 }
}
```

Tasks:

- [x] Create rule
- [x] Add entry to CHANGELOG.md
- [x] Review violations:
    - [x] Aerial - 7 violations
    - [x] Alamofire - no violations found
    - [x] Firefox - 3 violations
    - [x] Kickstarter - 14 violations
    - [x] Moya - 1 violation
    - [x] Nimble - no violations found
    - [x] Quick - no violations found
    - [x] Realm - 14 violations
    - [x] SourceKitten - 4 violations
    - [x] Sourcery - no violations found
    - [x] Swift - 96 violations
    - [x] WordPress - 2 violations

### CODE REVIEW

**Comment by ornithocoder** [comment]
I checked all the violations above and the only one bugging me is this one:

https://github.com/mozilla-mobile/firefox-ios/blob/3092d4d7843c9ec9fafa5f29e3d020ad7bbebaff/Client/Frontend/Browser/Punycode.swift#L145:17

**Comment by ornithocoder** [comment]
Review violations:

- [x] Aerial
- [x] Firefox
- [x] Kickstarter
- [x] Moya
- [x] Realm
- [x] SourceKitten
- [x] Swift

**Comment by ornithocoder** [comment]
Hi @jpsim @marcelofabri. This one is ready for review. Thanks! :-)

File: CHANGELOG.md [hunk 1]
```diff
@@ -76,6 +76,10 @@
   [Marcelo Fabri](https://github.com/marcelofabri)
   [#1504](https://github.com/realm/SwiftLint/issues/1504)
 
+* Add `multiline_parameters` opt-in rule that warns to either keep
+  all the parameters of a method or function on the same line, or one per line.
```
Comment (marcelofabri) [inline CHANGELOG.md]: requires two trailing spaces as described in CONTRIBUTING.md: https://github.com/realm/SwiftLint/blob/master/CONTRIBUTING.md#tracking-changes

Comment (ornithocoder) [inline CHANGELOG.md]: Oops. Foi mal. I completely forgot about that. Pushing the change right now.

Comment (ornithocoder) [inline CHANGELOG.md]: @marcelofabri fixed. Thanks for catching it.

**Comment by ornithocoder** [comment]
@marcelofabri thanks for catching the two trailing spaces. It should be fixed now.

**Pull request closed without merging**